### ISSUE TITLE
different behavior from mysql planner with a condition caused error.

### ISSUE THREAD

**Issue by wshwsh12** [pingcap/tidb#21625]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

CREATE TABLE ttt (
a int(10) unsigned,
b int(10) unsigned
);
insert into ttt values(1,10),(20,2);
select * from ttt where a-b>0 and a = 1 and b = 9;

### 2. What did you expect to see? (Required)

In MySQL 8.0
```
MySQL [test]>  select * from ttt where a-b>0 and a = 1 and b = 9;
ERROR 1690 (22003): BIGINT UNSIGNED value is out of range in '(1 - 9)'
```

### 3. What did you see instead (Required)

```
[tidb]> select * from ttt where a-b>0 and a = 1 and b = 9;
Empty set (0.001 sec)
[tidb]> desc select * from ttt where a-b>0 and a = 1 and b = 9;
+-------------+---------+------+---------------+---------------+
| id          | estRows | task | access object | operator info |
+-------------+---------+------+---------------+---------------+
| TableDual_5 | 1.60    | root |               | rows:0        |
+-------------+---------+------+---------------+---------------+
1 row in set (0.001 sec)
```

### 4. What is your TiDB version? (Required)
6cd3c6574

**Comment by wshwsh12** [comment]
Of course, it may be a MySQL bug. Because the condition is always false and it should get an empty result.

**Comment by zhangysh1995** [comment]
> Of course, it may be a MySQL bug. Because the condition is always false and it should get an empty result.

I think it is probably not a bug in MySQL. You see that `a` and `b` are unsigned int, and `a-b` compute negative numbers in a signed case but it will cause an underflow in an unsigned case, where you will get asuper big number as the result. It is reasonable that MySQL returned the out of range error.
